viewing_party/party.py

```diff
@@ -1,23 +1,132 @@
+import statistics
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
-
+# -----------------------------------------
 def create_movie(title, genre, rating):
-    pass
+    # create movie information
+    movie_info = {}
+    if title == None or genre == None or rating == None:
+        return None
+    movie_info['title'] = title
+    movie_info['genre'] = genre
+    movie_info['rating'] = rating
+    return movie_info 
+
+
+def add_to_watched(user_data, movie):
+    # add movie to what user has watched
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    # add movie to user's watchlist
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):

```

Comment:
To avoid iterating and modifying data at the same time, in this case, you could create one for loop with an if conditional that removes the movie from the watchlist and adds the movie to watched in one loop. As soon as we make a change we want to stop iterating. Here is a suggestion:
```python
def watch_movie(user_data, title):
    for movie in user_data["watchlist"]:
        if movie["title"] == title:
            user_data["watchlist"].remove(movie)
            user_data["watched"].append(movie)
            break
    return user_data
```